src/components/faqTemplates/ExpandableWithCategories.jsx

```diff
@@ -52,60 +15,55 @@ const TitleImg = styled.img`
 // FAQ Section with two columns with layout:
 // General    Logistics
 //            Teams & Projects
-const FaqSection = props => {
-  const [generalBin, setGeneralBin] = useState([])
-  const [logisticsBin, setLogisticsBin] = useState([])
-  const [teamsBin, setTeamsBin] = useState([])
+const ExpandableWithCategories = ({ faq, config }) => {
+  const [categorizedFaqMap, setFaqMap] = useState(new Map())
+
+  const categorizeFaq = faq => {
+    faq.forEach(({ category, question, answer }) => {
+      try {
+        const reducedFaq = { question, answer }
+
+        const currFaqList = categorizedFaqMap.get(category)
+        const updatedFaqList = currFaqList ? [reducedFaq, ...currFaqList] : [reducedFaq]
 
-  const distributeFaq = faq => {
-    faq.forEach(item => {
-      switch (item.category) {
-        case 'General':
-          setGeneralBin([...generalBin, item])
-          break
-        case 'Logistics':
-          setLogisticsBin([...logisticsBin, item])
-          break
-        case 'Teams & Projects':
-          setTeamsBin([...teamsBin, item])
-          break
-        default:
-          throw new Error(`FAQ category ${item.category} not found`)
+        const updatedMap = categorizedFaqMap.set(category, updatedFaqList)
+        setFaqMap(new Map(updatedMap))
+      } catch (e) {
+        throw new Error(e)
       }
     })
   }
 
   useEffect(() => {
     try {
-      const { faq } = props
-      distributeFaq(faq)
+      categorizeFaq(faq)
     } catch (e) {
       console.error(e)
     }
-  }, [props])
+  }, [faq])
 
   return (
-    <SectionContainer width={props.config.containerWidth} margin={props.config.containerMargin}>
+    <SectionContainer width={config.containerWidth} margin={config.containerMargin}>
       <TitleImg
-        src={props.config.titleImg}
-        alt={props.config.titleAlt}
-        marginBottom={props.config.marginBottomTitle}
+        src={config.titleImg}
+        alt={config.titleAlt}
+        marginBottom={config.marginBottomTitle}
       />
-      <Spacers height={props.config.titleBottomSpacing} />
+      <Spacers height={config.titleBottomSpacing} />
       <Columns>
         <Column>
           General
-          <FaqList list={generalBin} />
+          <ExpandableFaqList list={categorizedFaqMap.get('General') ?? []} />
```

Comment:
wonder if the ExpandableFaqList  component should handle the empty case? Then we won't have to null check every time we use the component?

Reply:
Sure, I'll pass the null check logic to `ExpandableFaqList`